sdk/core/azure-core/inc/azure/core/cryptography/hash.hpp

```diff
@@ -64,8 +64,11 @@ namespace Azure { namespace Core { namespace Cryptography {
      */
     void Append(const uint8_t* data, size_t length)
     {
+      // LCOV_EXCL_START
       _azure_ASSERT(data || length == 0);
```

Comment:
In the Embedded C SDK, we had a mechanism to test preconditions. Can we do something similar to these azure asserts? They are meaningful for correctness for the user, and hence having them covered by tests has value.

Reply:
I agree, it can be done.